pangeo_forge/cli.py

```diff
@@ -1,12 +1,47 @@
+import runpy
+
 import click
 
 
-@click.command()
+@click.group()
 @click.version_option()
 def main():
-    """Pangeo-Forge: A tool for building and publishing analysis ready datasets.
+    pass
+
+
+@click.command()
+@click.argument("pipeline", type=click.Path(exists=True))
+def lint(pipeline):
+    """
+    Check that the pipeline definition is valid. This does not run the pipeline.
     """
-    click.secho("Hello World!", fg="green")
+    result = runpy.run_path(pipeline)
+    missing = [key for key in ["pipeline", "flow"] if key not in result]
+    if missing:
+        click.echo(f"missing {missing}")
+    pipe = result["pipeline"]
+
+    pipe.flow.sorted_tasks()
+    pipe.flow.environment
+    pipe.flow.storage
+    pipe.flow.validate()
+    print("ok!")
+
+
+@click.command()
+@click.argument("pipeline", type=click.Path(exists=True))
+def register(pipeline):
```

Comment:
Should there be an option to specify _where_ to register the pipeline? In my head, this is separate from the pipeline itself.